pkg/util/errors/errors.go

```diff
@@ -4,17 +4,40 @@ import (
 	"net"
 	"net/url"
 	"os"
+	"strings"
 	"syscall"
 
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
 )
 
+// some common network errors that we check for
+var str2Syscall = map[string]syscall.Errno{
+	"connection reset by peer":  syscall.ECONNRESET,
+	"connection refused":        syscall.ECONNREFUSED,
+	"network is unreachable":    syscall.ENETUNREACH,
+	"no such file or directory": syscall.ENOENT,
+}
+
+func unwrapNetHTTPError(err error) (bool, error) {
+	if !strings.HasPrefix(err.Error(), "net/http: HTTP/1.x transport connection broken:") {
```

Comment:
what do you think about extracting that string to a variable `foo` so we can use `len(foo)` below instead of 47?